bebraven/platform
# Add persistent CSRF token cookie fix

### Description

Add [this Rails CSRF cookie bugfix](https://gist.github.com/killthekitten/b9a7b11530c44e788a31ec53e5ef0dad), to fix issues with CSRF tokens "expiring" after a Rails restart.

### Motivation and Context

Task: https://app.asana.com/0/1174274412967132/1201121531384087/f

In short, every daily Heroku restart, or re-deploy, all CSRF tokens were invalidated. That meant, if you had a project open before the restart and tried to submit it after the restart, it would fail. Now, CSRF tokens will be valid for a year instead. This will also fix any other page that relied on CSRF token validation for form submissions or AJAX requests.

See the Gist linked at the top of this PR for more context on the cause of this bug.

#### Steps to reproduce / testing

Without this fix in place:

1. In a Canvas course pointing at your local platformweb, as an enrolled Fellow, open a Project.
2. Restart your platformweb container.
3. Click Submit (or re-submit) in the Project.
4. See a red error message, get a 422 from platformweb.

With this fix in place:

1. In a Canvas course pointing at your local platformweb, as an enrolled Fellow, open a Project.
2. Restart your platformweb container.
3. Click Submit (or re-submit) in the Project.
4. See a green success message, get a 302 from platformweb.

Note: you may have to clear your platform cookies after this change is applied, before it takes effect.

### Types of changes




### Checklist:
